### DISCUSSION TITLE
It is possible to publish to "GitHub packages" using "GitHub Apps" generated token?

### DISCUSSION THREAD

**Discussion by heinzenberg** [community/community#28039] [Actions]
https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages

It is possible to publish packages in GitHub packages using **generated token from GitbHub Apps** instead of PAT!?

**Please remember:**
In order to generate token using GitHub Apps the process is:
https://docs.github.com/en/enterprise-cloud@latest/developers/apps/building-github-apps/authenticating-with-github-apps#http-based-git-access-by-an-installation

$ curl -i \
-H "Authorization: token YOUR_INSTALLATION_ACCESS_TOKEN" \
-H "Accept: application/vnd.github+json" \
https://api.github.com/installation/repositories

**Comment by dgholz** [comment]
> It is possible to publish packages in GitHub packages using **generated token from GitbHub Apps** instead of PAT!?

No. There's no API endpoints for publishing packages, so you have to use the language-specific registries, and they only accept a PAT (or a `GITHUB_TOKEN` generated as part of a workflow run (https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages#authenticating-to-github-packages)).

Even if there were API endpoints for publishing, GitHub Packages doesn't make any part of its API available to GitHub Apps.
